Model/ResourceModel/Magento/Product/CollectionBuilder.php

```diff
@@ -84,18 +91,21 @@ public function build()
      */
     public function withOnlyVisibleInSites()
     {
-        $this->collection->setVisibility($this->productVisibility->getVisibleInSiteIds());
+        $this->collection->addAttributeToFilter('visibility', ['neq' => ProductVisibility::VISIBILITY_NOT_VISIBLE]);
```

Comment:
Isn't this filter passed twice?

Reply:
Where?

Reply:
Is this used anywhere? https://github.com/Nosto/nosto-magento2/pull/755/files#diff-38d35079bd0e52a3ac1db621841fc1b4fe1e11eb2382ab66c5aebea0cb98ce27R48

Reply:
Removed